radixdlt-core/radixdlt/src/main/java/com/radixdlt/statecomputer/RadixEngineModule.java

```diff
@@ -104,8 +97,9 @@ PostParsedChecker checker(Set<PostParsedChecker> checkers) {
 		//   .toWindowedSet(initialValidatorSet, RegisteredValidatorParticle.class, p -> p.getAddress(), 2)
 		//   .build();
 
-		radixEngine.addStateReducer(new ValidatorsReducer(), true);
-		radixEngine.addStateReducer(new StakesReducer(), true);
+		radixEngine.addStateReducer(stakedValidatorsReducer, true);
+		//radixEngine.addStateReducer(new DeprecatedStakesReducer(), true);
+		radixEngine.addStateReducer(new InflationReducer(), "inflation", true);
```

Comment:
So, we can say that Reducers are "configuration" for ApplicationStateComputer?

We keep all of them in the RadixEngine. Providing the name is defined, it's going to be the key to get it from the map.

Depending on particle spin we use output or input function defined in the Reducer.

The processCheckSpin is not only interpreting the spin, but actually provides execution context for any function.

Reply:
I think sounds about right. Reducers map groups of substate to higher level state constructs, `stakedValidatorsReducer` being the most important as it maps registered validators and stake to `BFTValidatorSet` which is used for consensus. `InflationReducer` isn't really used much except for testing.